ext/civiimport/Civi/Api4/Event/Subscriber/ImportSubscriber.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+namespace Civi\Api4\Event\Subscriber;
+
+use Civi;
+use Civi\Core\HookInterface;
+use CRM_Core_DAO;
+use Civi\Core\Event\GenericHookEvent;
+use CRM_Utils_Rule;
+use Civi\Api4\Import;
+
+/**
+ * Listening class that registers each Import table as an entity.
+ */
+class ImportSubscriber implements HookInterface {
+
+  /**
+   * @return array
+   */
+  public static function getImportTables(): array {
+    // We need to avoid the api here as it
+    $tables = CRM_Core_DAO::executeQuery('
+       Select `id`, `metadata`, `name`, `job_type`, `created_id`
+       FROM civicrm_user_job
+       -- As of writing expires date is probably not being managed
+       -- it is intended to be used to actually purge the record in
+       -- a cleanup job so it might not be relevant here & perhaps this will
+       -- be removed later
+       WHERE (expires_date IS NULL OR expires_date > NOW())
+       -- this is a short-cut for looking up if they are imports
+       -- it is a new convention, at best, to require anything
+       -- specific in the job_type, but it saves any onerous lookups
+       -- in a function which needs to avoid loops
+       AND job_type LIKE "%import"
+         -- also more of a feature than a specification - but we need a table
+         -- to do this pseudo-api
+       AND metadata LIKE "%table_name%"');
+    $importEntities = [];
+    while ($tables->fetch()) {
+      $tableName = json_decode($tables->metadata, TRUE)['DataSource']['table_name'];
+      if (!CRM_Utils_Rule::alphanumeric($tableName) || !CRM_Core_DAO::singleValueQuery('SHOW TABLES LIKE %1', [
+          1 => [
+            $tableName,
+            'String'
+          ]
+        ])) {
+        continue;
+      }
+      $importEntities[$tables->id] = $tableName;
+    }
+    return $importEntities;
+  }
+
+  /**
+   * Register each valid import as an entity
+   *
+   * @param GenericHookEvent $event
+   *
+   * @noinspection PhpUnused
+   * @noinspection PhpUndefinedFieldInspection
+   */
+  public static function on_civi_api4_entityTypes(GenericHookEvent $event): void {
+    $importEntities = self::getImportTables();
+    foreach ($importEntities as $userJobID => $tableName) {
+      $event->entities['Import_' . $userJobID] = [
+        'name' => 'Import_' . $userJobID,
+        'title' => ts('Import') . '_' . $userJobID,
+        'title_plural' => ts('Import') . $userJobID,
```

Comment:
This is kinda ugly and not the right way to use `ts()`, is there no user-facing label for these temp tables?